salt/modules/slsutil.py

```diff
@@ -246,3 +247,130 @@ def deserialize(serializer, stream_or_string, **mod_kwargs):
+        use the ``blockstart`` and ``blockend`` options.
+    :param borderchar: The character to use in the top and bottom border of
+        the comment box. Must be a single character.
+    :param blockstart: The character sequence to use at the beginning of a
+        block comment. Should be used in conjunction with ``blockend``
+    :param blockend: The character sequence to use at the end of a
+        block comment. Should be used in conjunction with ``blockstart``
+    :param title: The first field of the comment block. This field appears
+        centered at the top of the box.
+    :param text: The second filed of the comment block. This field appears
+        left-justifed at the bottom of the box.
+    :param newline: Boolean value to indicate whether the comment block should
+        end with a newline. Default is ``False``.
+
+    This banner can be injected into any templated file, for example:
+
+    .. code-block:: jinja
+
+        {{ salt['slsutil.banner'](width=120, commentchar='//') }}
+
+    The default banner:
+
+    .. code-block:: none
+
+        ########################################################################
+        #                                                                      #
+        #              THIS FILE IS MANAGED BY SALT - DO NOT EDIT              #
+        #                                                                      #
+        # The contents of this file are managed by Salt. Any changes to this   #
+        # file may be overwritten automatically and without warning.           #
+        ########################################################################
+    '''
+
+    if title is None:
+        title = 'THIS FILE IS MANAGED BY SALT - DO NOT EDIT'
+
+    if text is None:
+        text = ('The contents of this file are managed by Salt. '
+                'Any changes to this file may be overwritten '
+                'automatically and without warning')
+
+    # Set up some typesetting variables
+    lgutter = commentchar.strip() + ' '
+    rgutter = ' ' + commentchar.strip()
+    textwidth = width - len(lgutter) - len(rgutter)
+    border_line = commentchar + borderchar[:1] * (width - len(commentchar) * 2) + commentchar
+    spacer_line = commentchar + ' ' * (width - len(commentchar) * 2) + commentchar
+    wrapper = textwrap.TextWrapper(width=(width - len(lgutter) - len(rgutter)))
+    block = list()
+
+    # Create the banner
+    if blockstart is not None:
+        block.append(blockstart)
+    block.append(border_line)
+    block.append(spacer_line)
+    for line in wrapper.wrap(title):
+        block.append(lgutter + line.center(textwidth) + rgutter)
+    block.append(spacer_line)
+    for line in wrapper.wrap(text):
+        block.append(lgutter + line + ' ' * (textwidth - len(line)) + rgutter)
+    block.append(border_line)
+    if blockend is not None:
+        block.append(blockend)
+
+    # Convert list to multiline string
+    result = '\n'.join(block)
```

Comment:
For better multi-platform support we should probably use `os.linesep` here instead

Reply:
That suggestion sounded right to me, but the [documentation](https://docs.python.org/3/library/os.html#os.linesep) is making me think not:

*Do not use os.linesep as a line terminator when writing files opened in text mode (the default); use a single '\n' instead, on all platforms.*

It seems that Python converts `\n` to `os.linesep` when it writes out the file.

Reply:
The files aren't ultimately opened in text mode, though. That's the thing. :)